src/uris/deepLinks/deepLinkService.ts

```diff
@@ -119,35 +143,61 @@ export class DeepLinkService implements Disposable {
 		return undefined;
 	}
 
-	private async processDeepLink(): Promise<void> {
+	private async showOpenRepoPrompt(): Promise<DeepLinkRepoOpenAction> {
+		const result = await window.showInformationMessage(
+			`No matching repository found. Please choose an option to open the repository.`,
+			{ modal: true },
+			{ title: DeepLinkRepoOpenAction.OpenInCurrentWindow },
```

Comment:
We can extend/add extra props here, so we could add an `action` prop or something that points directly to `OpenWorkspaceLocation`, and it could be `undefined` or missing on `Cancel`

Reply:
Done in commit: [Improvements from feedback](https://github.com/gitkraken/vscode-gitlens/pull/2503/commits/9e01ae826ec49b794057aa9570ec3df54ef0042d)